core/server/server.go

```diff
@@ -49,11 +51,19 @@ func NewCoreConfig(log logr.Logger, cfg *rest.Config, clusterName string) CoreSe
 }
 
 func NewCoreServer(cfg CoreServerConfig) pb.CoreServer {
+	ctx := context.Background()
+
 	cfgGetter := kube.NewImpersonatingConfigGetter(cfg.RestCfg, false)
+	k8s := kube.NewDefaultClientGetter(cfgGetter, cfg.clusterName)
```

Comment:
we should avoid using this Initializer since it only adds unnecessary complexity.

```go

import (
    "sigs.k8s.io/controller-runtime/pkg/client"
)

cl, err := client.New(cfg.RestCfg, client.Options{})
```

The snippet above should do the trick